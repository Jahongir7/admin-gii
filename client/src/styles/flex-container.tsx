import styled from 'styled-components';

export type FlexContainerProps = {
  wrap?: 'wrap' | 'nowrap' | 'wrap-reverse' | 'nowrap-reverse' | 'initial' | 'inherit';
  justify?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'initial' | 'inherit';
  align?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'initial' | 'inherit';
  [key: string]: any;
}

export const FlexContainer = styled.div<FlexContainerProps>`
  display: flex;
  flex-wrap: ${props => props.wrap};
  justify-content: ${props => props.justify};
  align-items: ${props => props.align};
`;

FlexContainer.defaultProps = {
  justify: 'initial',
  align: 'initial',
  wrap: 'initial',
};
