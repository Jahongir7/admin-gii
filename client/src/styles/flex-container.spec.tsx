import { customRender } from '@test-utils';
import { screen } from '@testing-library/react';
import { FlexContainer } from './flex-container';

test('Should render FlexContainer component', () => {
  const { asFragment } = customRender(<FlexContainer />);

  expect(asFragment).toMatchSnapshot();
});

test('FlexContainer should get justify and align in props', () => {
  const JUSTIFY = 'center';
  const ALIGN = 'center';
  const WRAP = 'wrap';

  customRender(
    <FlexContainer
      data-testid='tested-flex-container'
      justify={JUSTIFY}
      align={ALIGN}
      wrap={WRAP}
    />
  );
  const elem = screen.getByTestId('tested-flex-container');
  const styles = window.getComputedStyle(elem);
  expect(styles.justifyContent).toBe(JUSTIFY);
  expect(styles.alignItems).toBe(ALIGN);
  expect(styles.flexWrap).toBe(WRAP);
});
