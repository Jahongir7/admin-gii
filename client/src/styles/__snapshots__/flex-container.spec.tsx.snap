// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FlexContainer component 1`] = `[Function]`;
